keras/engine/training.py

```diff
@@ -983,7 +983,8 @@ def _standardize_user_data(self, x, y,
         sample_weights = [standardize_weights(ref, sw, cw, mode)
                           for (ref, sw, cw, mode)
                           in zip(y, sample_weights, class_weights, self.sample_weight_modes)]
-        check_array_lengths(x, y, sample_weights)
+        if check_batch_dim:
+            check_array_lengths(x, y, sample_weights)
```

Comment:
In the training process for domain adaptation, the input batch contains ```batch_size/2``` samples from the **source** domain and ```batch_size/2``` samples from the **target** domain. Per the model (added picture in PR), the full batch is used in the left branch to do "domain classification" whereas the first half of the batch is sliced out feature extraction in the right branch for the **source** domain.

The purpose of this is to bypass a sanity check in Keras by propagating a ```kwarg``` from ```train_on_batch``` and allow unequal batch lengths for the X and y arguments.